### ISSUE TITLE
Help with writing Documentation

### ISSUE THREAD

**Issue by power-fungus** [llogiq/mutagen#140]
Hi, I am very interested in this project and am starting to read the code to better understand its structure. At the end of this process, I would like to write some documents explaining the architecture and codebase of this project.

However, I fail to currently understand some critical parts of the project. @llogiq, could you help out with a few questions?
* why is there a split between the `plugin` and the main `mutagen` library? What are the interface boundaries between these 2?
* some time ago there was a huge `build.rs`. What was that for and where has the code been moved to?

I understand that these questions may not be trivial to answer, but even brief hints to where to look would help me writing some architecture documents. Thanks ;)

**Comment by llogiq** [comment]
To answer your first question, the main `mutagen` library contains vital support functions to reduce the amount of code needed for the mutations. The `plugin` is a `proc_macro_attribute` to introduce the mutations in the annotated code. I must admit that I'd like to change the API by re-exporting `mutagen_plugin::mutate` from `mutagen` (so that the plugin crate becomes a mere implementation detail), but the plugin relies on the library, so this would create a circular dependency.

As for the second question, I had used the `build.rs` scripts to get around earlier problems with using macros to create the code. Those problems are no longer there when using `syn` (as opposed to the old plugin interface), so I changed them to plain macro_rules definitions in both `mutagen` and `mutagen_plugin`.
